lib/Dialect/TritonGPU/Transforms/Pipeliner/PipeliningUtility.cpp

```diff
@@ -379,3 +379,52 @@ ttg::MemDescType mlir::triton::getBufferViewType(ttg::MemDescType allocTy) {
                                sharedMemorySpace, /*mutableMemory=*/true,
                                /*allocShape=*/allocTy.getAllocShape());
 }
+
+ttg::SharedEncodingTrait mlir::triton::getSharedEncoding(RankedTensorType ty) {
+  auto ctaLayout = ttg::getCTALayout(ty.getEncoding());
+  auto order = ttg::getOrder(ty);
+  // Use generic layout. This won't be optimal for 2D tensors.
+  return ttg::SwizzledSharedEncodingAttr::get(ty.getContext(), 1, 1, 1, order,
+                                              ctaLayout);
+}
```

Comment:
@pawelszczerbuk is planning to re-work this part to move it after the pipeliner/warp specialization, that should be a nice clean up for both

Reply:
Interesting! How exactly is that supposed to work? Are we going to be bufferizing without picking a shared layout?